### PR TITLE
[FORMATS-170] Add sampleId to Feature record

### PR DESCRIPTION
**Pull request by akmorrow13**

### CODE REVIEW

File: src/main/resources/avro/bdg.avdl [hunk 1]
```diff
@@ -1165,6 +1165,12 @@ record Feature {
    */
   union { null, string } featureId = null;
 
+
+  /**
+   Identifier for this sample.
+   */
+  union { null, string } sampleId = null;
```
Comment (akmorrow13) [inline src/main/resources/avro/bdg.avdl]: @heuermh do you think this should be an Avro Sample or just a string sampleId?

Comment (heuermh) [reply src/main/resources/avro/bdg.avdl]: Elsewhere we use `sampleId`, with the expectation that users will join on sampleId against  metadata `samples` &rarr; `Sample.sampleId`.

Comment (heuermh) [reply src/main/resources/avro/bdg.avdl]: Yup.  I'd suggest updating the doc string to something like
```java
  /**
   Sample identifier for this feature. Join with Sample.sampleId for sample metadata.
   sampleId tag in GFF3.
  */
```
This matches the style used in #108.

**Comment by heuermh** [comment]
Propose to milestone version 0.11.4.

**Comment by heuermh** [comment]
Thank you, @akmorrow13

**Pull request merged**